bitcoin/src/blockdata/fee_rate.rs

```diff
@@ -209,8 +212,11 @@ mod tests {
 
     #[test]
     fn checked_weight_mul_test() {
-        let weight = Weight::from_wu(10);
-        let fee: Amount = FeeRate(10).checked_mul_by_weight(weight).expect("expected Amount");
+        let weight = Weight::from_vb(10).unwrap();
```

Comment:
Why change this to `from_vb` instead of `from_wu`?

Suggest:
```
let weight = Weight::from_wu(10); 
let fee: Amount = FeeRate(10).checked_mul_by_weight(weight).expect("Amount not to overflow");
```

Also, the current expect messages suck (sorry.  It should say something like "Amount not to overflow".

Reply:
Same reason as [for this change](https://github.com/rust-bitcoin/rust-bitcoin/pull/2182/files/0c56131819bd04270bd14e6da1179645798c2e14#r1389671792): intuition for the numbers of the test.

Reply:
There's already two acks, and I think it's good to get this fixed asap, so no more changes needed.  Although, I still feel like `from_vb()` makes this test more confusing.  Keeping it all as Weight Units:

```
    #[test]
    fn checked_weight_mul_test() {
        let weight = Weight::from_wu(10);
        let fee: Amount = FeeRate(10).checked_mul_by_weight(weight).expect("expected Amount");
        assert_eq!(Amount::from_sat(1), fee);

        let fee = FeeRate(10).checked_mul_by_weight(Weight::MAX);
        assert!(fee.is_none());
    }
```

since we start at 10 Weight units, and then add 1_000,  inside checked_mul, this results in 1,000 Wu * 10 sats / 1,000 Wu = 1 sat.  The multiplication by 4 when using `from_vb()` doesn't add anything to this test and is diff noise.